joomla/joomla-cms
# [4.0] Media fields accessibility - second try

Pull Request for Issue #27571 .

Please see https://github.com/joomla/joomla-cms/pull/27650 and https://github.com/joomla/joomla-cms/pull/27712#issuecomment-580956349  for test instructions.

In this Pull Request I tried to implement a suggestion from this [comment](https://github.com/joomla/joomla-cms/pull/27650#issuecomment-579773758
) - I use the Subform Field.


Edit: 10th of February
Test Instructions - You do NOT need composer or npm

1. Create a Custom Field of type media and see, that this field did not include an alternative text in front end. I named this custom field `old`.
2. Apply this patch and create a new custom field of type media. See hat all parameters, for example directory, are working. I named this custom field `new`.
3. See, that the custom fields `new` is still shown correctly in front end. ~When you edit an article, the text field for the alternative text is not show for the `old` custom field at the begin. But when you delete the image temporarily the text field appear and you can use it.~
4. Use the new form field regardless of custom field
a) Add 
```
		<field
			name="image2"
			type="accessiblemedia"
			label="COM_CONTACT_FIELD_PARAMS_IMAGE_LABEL"
			directory="banners"
		/>

```
somewhere for example in line 276 of `/administrator/components/com_contact/forms/contact.xml`.
b) If you use my example add

		`<?php echo $this->form->renderField('image2'); ?>`

in `/administrator/components/com_contact/tmpl/contact/edit.php` line 55.
c) See that you can use the field.
